src/gtp_path.erl

```diff
@@ -168,10 +168,16 @@ handle_event({call, From}, {bind, Pid, RestartCounter}, _State, Data0) ->
 
 handle_event({call, From}, {unbind, Pid}, _State, Data0) ->
     Data = unregister(Pid, Data0),
-    {keep_state, Data, [{reply, From, ok}]};
+    case Data of
+	#{echo_timer := stopped} ->
```

Comment:
The echo_timer state could IMHO be part of the overall state.

In the conversion to gen_statem, make state a record with a single field `state` (UP/DOWN). And in this cset, add the the echo_timer state into that state.

Reply:
@RoadRunnr 
Do you mean something like: 
-record(state, {state :: {atom(), atom()}}). %  { ['UP' | 'DOWN'], [stopped | echo_to_send | awaiting_response])
Some thing like this will certainly be easier pattern match on the gen_statem state field. 
Can expand the tuple to further add future timers for path maintenance.

Reply:
more like:
```Erlang
record(state, {peer :: 'UP' | 'DOWN',
               echo :: idle | pending)).
```
that only a idea, the real implementation could need different values.